controllers/azureidentity_controller.go

```diff
@@ -96,20 +111,38 @@ func (r *AzureIdentityReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		))
 	defer span.End()
 
+	// identityOwner is the resource that created the identity. This could be either an AzureCluster or AzureManagedControlPlane (if AKS is enabled).
+	// check for AzureManagedControlPlane first and if it is not found, check for AzureManagedControlPlane.
+	var identityOwner interface{}
+
 	// Fetch the AzureCluster instance
 	azureCluster := &infrav1.AzureCluster{}
+	identityOwner = azureCluster
 	err := r.Get(ctx, req.NamespacedName, azureCluster)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
+	if err != nil && apierrors.IsNotFound(err) {
+		if feature.Gates.Enabled(feature.AKS) {
```

Comment:
instead of doing this check, we could probably use an unstructured client to lookup whatever GVK/object we get, pull `spec.identity` and drop the feature gate check + error handling.

fine to leave for follow up imo